cpp/src/arrow/util/thread_pool.cc

```diff
@@ -386,7 +405,8 @@ struct ThreadPool::State {
   std::list<std::thread> workers_;
   // Trashcan for finished threads
   std::vector<std::thread> finished_workers_;
-  std::deque<Task> pending_tasks_;
+  std::priority_queue<QueuedTask> pending_tasks_;
+  uint64_t spawned_tasks_count = 0;
```

Comment:
Nit: naming
```suggestion
  uint64_t spawned_tasks_count_ = 0;
```

Reply:
I just realised I forgot about this and I also should have noticed the naming convention, when I originally wrote it. Pushing a correction now